src/pgReal.ts

```diff
@@ -98,7 +99,10 @@ export async function withClient<T = void>(
   parsedSettings: ParsedSettings,
   callback: (pgClient: PoolClient, context: Context) => Promise<T>,
 ): Promise<T> {
-  const details = getPoolDetailsFromConnectionString(connectionString);
+  const details = getPoolDetailsFromConnectionString(
+    connectionString,
+    parsedSettings.logger,
```

Comment:
In general I'd rather `parsedSettings` was passed around "whole" - that way if functions later need something else from that object we don't have to add additional parameters.
```suggestion
    parsedSettings,
```